gradle/java.gradle

```diff
@@ -96,6 +96,25 @@ plugins.withId('java') {
         def jvmArgsCopy = jvmArgs
         jvmArgsCopy.add("-XX:MaxDirectMemorySize=4g")
         jvmArgs = jvmArgsCopy
+        def outputCache = new LinkedList<String>()
+
+        beforeTest { descriptor ->
+            logger.lifecycle("\nRunning test: " + descriptor.getClassName() + " > " + descriptor.getDisplayName());
+            outputCache.clear()
+        }
+
+        onOutput { descriptor, event ->
+            outputCache.add(event.getMessage())
+            while (outputCache.size() > 300) outputCache.remove()
+        }
+
+        afterTest { descriptor, result ->
+            if (result.resultType == TestResult.ResultType.FAILURE && outputCache.size() > 0) {
+                println()
+                println(" Output of ${descriptor.getClassName()}.${descriptor.getDisplayName()}:")
```

Comment:
Shouldn't this be using the `logger` too?

Reply:
Also, my understanding is that the following lines will already ensure that any exceptions for failed tests are logged. Curious, do we still need these explicit statements here?

https://github.com/pravega/pravega/blob/9efbe1e1fcd8d074ceafda91ff67f918c245cb60/gradle/java.gradle#L89-L92

Reply:
Thanks for reviewing the PR @ravisharda 

>Shouldn't this be using the logger too?

Using a `logger` at `INFO `level will enforce all the gradle commands to be use the `--info` option e.g: .`/gradlew build --info`. Also at INFO the log output from gradle will clutter the console with other information messages, which is not desireable. 
I do not want to use the `logger.lifecycle(...)` API since it should be used for progress information messages, ref: https://docs.gradle.org/current/userguide/logging.html

> Regarding options testLogging.showCauses, showExceptions...

`showExceptions `and `showCauses` settings are used to log if there are any exceptions during the execution of the test, even though the test is successful. `showStackTraces `ensures not just the exception but its stacktrace is logged during the run. Hence these options are required.

Reply:
On second look, it looks like the code is trying to log stdout ad stderr for failed tests and failed tests alone. I have to say I don't like the imperative code needed to achieve this and also the fact that each passing test ends up having two lines in the console output. 

Should we log stdout and stderr in all cases, instead?

Reply:
As discussed, I have impoved the output of the gradle task.
Also, since the thread leak detector exits the gradle test worker jvm with exit code 99 there is no way for the gradle daemon to get this information as an event.